backend/src/db/migrations/20230826133720_add_role_id_to_users_table.ts

```diff
@@ -0,0 +1,42 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = {
+  USERS: 'users',
+  USER_ROLES: 'user_roles',
+} as const;
+
+const COLUMN_NAME = {
+  ROLE_ID: 'role_id',
+  ID: 'id',
+} as const;
+
+const USER_ROLE_KEY = 'user';
+
+type UserRow = {
+  id: number;
+};
+
+async function up(knex: Knex): Promise<void> {
+  const userRole: UserRow = (await knex(TABLE_NAME.USER_ROLES)
+    .where({ key: USER_ROLE_KEY })
+    .select(COLUMN_NAME.ID)
+    .first()) as UserRow;
```

Comment:
Just include UserRow as a generic to .select<> without type assertion

Reply:
@v0ldemar01  then userRole might be of type undefined, what am I supposed to return in that case?

Reply:
Let's proceed with such approach:
```js
const userRole = await knex(TABLE_NAME.USER_ROLES)
    .select<UserRow>(COLUMN_NAME.ID)
    .where({ key: USER_ROLE_KEY })
    .first();

  await knex.schema.alterTable(TABLE_NAME.USERS, (table) => {
    table
      .integer(COLUMN_NAME.ROLE_ID)
      .unsigned()
      .notNullable()
      .references(COLUMN_NAME.ID)
      .inTable(TABLE_NAME.USER_ROLES)
      .defaultTo((userRole as UserRow).id);
  });
```

Reply:
Yes, ok, changed it